packages/terra-clinical-item-collection/src/_ItemCollectionUtils.jsx

```diff
@@ -0,0 +1,101 @@
+const KEYCODES = {
+  ENTER: 13,
+  SPACE: 32,
+};
+
+/**
+ * This function ensures the correct elements are provided to create a consistent item view layout. To handle a
+ * potential missing required accessory element, a boolean is provided to indicate if start accessory space is needed.
+ *
+ * @ param {childElement} obj - The react child props that will compose an item view list item.
+ * @ param {requiredElements} obj - The elements expected to be displayed.
+ */
+function addMissingListElements(childElements, requiredElements) {
```

Comment:
I'll take another stab at this! I know I was confusing last time!

In the use of this method (https://github.com/cerner/terra-clinical/pull/221/files#diff-a8fa6bc226c2f3aa668a0e94d6f34228R35), I see that childElements corresponds to the props of the child, and requiredElements corresponds to props of the ListView. Given that, I think these should be renamed to childProps and requiredProps. Personally though, I think requiredProps should be parentProps, as it looks like it's taking props from the parent ListView.

Reply:
^ I see what you are saying here. I can update these to be childProps.

I also see what you mean about `requiredElements` as well. I wanted the concept of the requiredElements obj as these are the pieces needed to create a consistent layout. I am open to a different name, but parentProps to me would be more confusing and not a true name to represent what is being passed. When working through the code as requiredProps are created by the Item Collection and the same object is passed to the _tableView and _ListView and both component pass requiredElements to Utils.

Reply:
the documentation says the first param is props and the next contains elements, so we should just change the first params name to childProps

Reply:
I see what requiredElements are. It maps directly to the prop types of the listView. I like the structure it's going for with requiredElements, but I don't know if element is the best word. When I think element, I think of an html node or react component. For that prop, perhaps it's better to say requiredProps or requiredAttributes?

Reply:
I see what requiredElements are now. In that case, I would recommend childProps and requiredElements.